pkg/commands/menu/menu.go

```diff
@@ -111,24 +138,102 @@ func (h *MenuCommand) SayMenu(s *discordgo.Session, i *discordgo.InteractionCrea
 	var selectedCampus = i.ApplicationCommandData().Options[0].Value.(string)
 
 	data := GetSiteContent(selectedCampus)
+	if len(data) == 0 {
+		err := s.InteractionRespond(i.Interaction, &discordgo.InteractionResponse{
+			Type: discordgo.InteractionResponseChannelMessageWithSource,
+			Data: &discordgo.InteractionResponseData{
+				Content: "That campus does not have a menu for this week yet!",
+			},
+		})
 
-	currentMenu := []MenuData{}
-	for _, item := range data {
-		// if is today or after today
-		if item.Curdate.After(time.Now()) || item.Curdate.Day() == time.Now().Day() {
-			currentMenu = append(currentMenu, item)
+		if err != nil {
+			log.Println(err)
 		}
+		return
 	}
 
-	embeds := []*discordgo.MessageEmbed{}
-	for _, item := range currentMenu {
-		e := embed.NewEmbed()
-		e.Title = item.Curdate.Format("Monday")
-		for _, item := range item.Items {
-			e.AddField(item.Category.NameEN, item.ShortDescriptionEN)
+	ndata := data[0]
+	pdata := ndata.(map[string]interface{})
+
+	var items map[string]interface{}
+	var startDate time.Time
+
+	// retrieve list of categories and the startdate
+	for k, v := range pdata {
+		switch k {
+		case "items":
+			items = v.(map[string]interface{})
+		case "startdate":
+			startDate, _ = time.Parse(time.RFC3339, v.(string))
```

Comment:
we should check if this cast is valid
```suggestion
                        dateString, ok := v.(string)
                        if !ok {
                            break
                        }
			startDate, _ = time.Parse(time.RFC3339, dateString)
```